dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java

```diff
@@ -670,6 +670,11 @@ private DependResult isTaskDepsComplete(String taskName) {
         }
 
         TaskNode taskNode = dag.getNode(taskName);
+        // condition node directly return success
+        if (taskNode.isConditionsTask()) {
+            return DependResult.SUCCESS;
```

Comment:
good job, but condition task also need check all the depend nodes complete, so it cannot directly return 'success' in here.

Reply:
sorry, I didn't think it through. You are right, it needs to check all depend nodes. **So, the three lines here is not necessary.**